src/single.ts

```diff
@@ -138,10 +143,24 @@ export class RegionRatelimit extends Ratelimit<RegionContext> {
     return r
     `;
 
-    return async function (ctx: RegionContext, identifier: string) {
+    const rateScript = `
+    local key     = KEYS[1]
+    local incr_by  = ARGV[1]
+    local window  = ARGV[2]
+    
+    local r = redis.call("INCRBY", key, incr_by)
+    if r == incr_by then
+    -- The first time this key is set, the value will be 1.
+    -- So we only need the expire command once
+    redis.call("PEXPIRE", key, window)
+    end
+    
+    return r
+    `;
+
+    return async function (ctx: RegionContext, identifier: string, rate?: number) {
```

Comment:
Let's change `rate` to `requestCost` or something even better. Because this is the cost of each request when we set `customLimit`

Reply:
Got that